app/api/graph/[graph]/route.ts

```diff
@@ -0,0 +1,33 @@
+import { NextRequest, NextResponse } from "next/server";
+import { getServerSession } from "next-auth/next";
+import authOptions, { getConnection } from "../../auth/[...nextauth]/options";
+
+// eslint-disable-next-line import/prefer-default-export
+export async function DELETE(request: NextRequest, { params }: { params: { graph: string } }) {
+
+    const session = await getServerSession(authOptions)
+    const id = session?.user?.id
+    if (!id) {
+        return NextResponse.json({ message: "Not authenticated" }, { status: 401 })
+    }
+
+    const client = await getConnection(session.user)
+    if (!client) {
+        return NextResponse.json({ message: "Not authenticated" }, { status: 401 })
+    }
+
+    const graphId = params.graph;
+
+    try {
+        if (graphId) {
+            
+            const graph = client.selectGraph(graphId);
+
+            graph.delete()
```

Comment:
delete is async method need to call await

```suggestion
            await graph.delete()
```